### ISSUE TITLE
BulletList delete Mention

### ISSUE THREAD

**Issue by Morgbn** [ueberdosis/tiptap#2416]
### What’s the bug you are facing?

If a paragraph node starts by a mention, and then I type `*` , `-`  or `+`, a bullet list is created, and the previous mention is deleted.

### How can we reproduce the bug on our side?

On the [codesandbox](https://codesandbox.io/s/tiptap-issue-template-forked-ox987?file=/src/components/Tiptap.vue), add a new line, use @ to mention a user, type `*` , `-`  or `+`.

### Can you provide a CodeSandbox?

https://codesandbox.io/s/tiptap-issue-template-forked-ox987?file=/src/components/Tiptap.vue

### What did you expect to happen?

How can I prevent this behavior ?

### Anything to add? (optional)

![ezgif-7-0d4604a5da](https://user-images.githubusercontent.com/25689856/150516454-16a26d88-bd42-45f1-a646-cf785e48d070.gif)

### Did you update your dependencies?

- [X] Yes, I’ve updated my dependencies to use the latest version of all packages.

### Are you sponsoring us?

- [ ] Yes, I’m a sponsor. 💖

**Comment by philippkuehn** [comment]
Hmm, this is a regression from [this patch](https://github.com/ueberdosis/tiptap/commit/ded6f9618fb588f2c259edc9bc77415f8723431c) to fix [this issue](https://github.com/ueberdosis/tiptap/issues/2302). Not sure about how to solve both issues right now.

**Comment by floriankrueger** [comment]
Hi @philippkuehn, our customers are facing the same issue.

Since we customised the @mention extension quite a bit I initially thought I broke it but I could reproduce it on [a vanilla implementation of the editor](https://codesandbox.io/s/elastic-burnell-pmd7py) (like @Morgbn did).

Some rough thoughts: 

It seems as if the component that processes the input rules doesn't take the @mention node into consideration at all. Otherwise the regex `/^\s*([-+*])\s$/` (from the [bullet list](https://github.com/ueberdosis/tiptap/blob/c9eb6a6299796450c7c1cfdc3552d76070c78c65/packages/extension-bullet-list/src/bullet-list.ts#L19)) wouldn't match, since it's specifically asking for a dash, plus or asterisk on a new line. But there is something else in between: the mention. I'm unclear on which level of presentation that regex is applied, probably some kind of plain text, but it seems to me that the regex doesn't "see" the mention node. Couldn't we try to give the mention node a textual representation on this level?

Are there any plans to look into this? I can also do that, but I would probably need some help to get started here 🙂
